code/db.c

```diff
@@ -3402,16 +3402,18 @@ int number_range(int from, int to)
 	return from + number;
 }
 
-/*
- * Generate a percentile roll.
- */
+///
+/// Generate a percentile roll.
+///
 int number_percent(void)
 {
-	int percent;
+	// int percent;
+
+	// while ((percent = number_mm() & 127) > 99);
 
-	while ((percent = number_mm() & (128 - 1)) > 99);
```

Comment:
Are we sure we want to scrap the number generator? I mean it's a pretty valid multi-sampler (see below). My initial hunch was wrong. Though I'm not sure why they only include `1 - 99` as valid percentages. It maybe the fact the code as it stands can't handle the `0` and `100` as percentages.

```
#include <iostream>

int main() {
    for(int i = 0; i < 10; i++) {
        int r = rand();
        std::cout << r << std:: endl;
        int rr = r >> 6;
        std::cout << rr << std:: endl;
        int rrr = rr & (128 - 1);
        std::cout << rrr << std:: endl;
    }
  
    return 0;
}
```

```
$ ./main
1804289383
28192021
21
846930886
13233295
15
1681692777
26276449
97
1714636915
26791201
33
1957747793
30589809
113
424238335
6628723
115
719885386
11248209
81
1649760492
25777507
99
596516649
9320572
124
1189641421
18588147
115
```

Reply:
Honestly I think you're right to question it. I'll roll this portion back and add in a seed initializer for the tests. The problem I had during testing was the number always returned 1.  Better to add tests to validate it works rather than bypass.